font/src/directwrite/mod.rs

```diff
@@ -212,3 +306,57 @@ impl ::std::fmt::Display for Error {
         }
     }
 }
+
+fn em_size(size: Size) -> f32 {
+    size.as_f32_pts() * (96.0 / 72.0)
+}
+
+fn encode_utf16(c: char) -> Vec<u16> {
+    let mut buf = vec![0u16; c.len_utf16()];
+    c.encode_utf16(&mut buf);
+    buf
+}
+
+impl From<Weight> for FontWeight {
+    fn from(other: Weight) -> FontWeight {
+        match other {
+            Weight::Bold => FontWeight::Bold,
+            Weight::Normal => FontWeight::Regular,
+        }
+    }
+}
+
+impl From<Slant> for FontStyle {
+    fn from(other: Slant) -> FontStyle {
+        match other {
+            Slant::Oblique => FontStyle::Oblique,
+            Slant::Italic => FontStyle::Italic,
+            Slant::Normal => FontStyle::Normal,
+        }
+    }
+}
+
+fn get_current_locale() -> String {
+    let mut buf = vec![0u16; LOCALE_NAME_MAX_LENGTH];
+    let len = unsafe { GetUserDefaultLocaleName(buf.as_mut_ptr(), buf.len() as i32) as usize };
+
+    // len includes null byte, we don't need that in Rust strings
+    OsString::from_wide(&buf[..len - 1]).into_string().expect("Locale not valid unicode")
+}
+
+/// Information needed by dwrote's TextAnalysisSource to find a fallback font
+/// for a run of text. See `get_fallback_font`.
+struct TextAnalysisSourceData {
+    locale: String,
+    length: u32,
+}
+
+impl dwrote::TextAnalysisSourceMethods for TextAnalysisSourceData {
```

Comment:
```suggestion
impl TextAnalysisSourceMethods for TextAnalysisSourceData {
```

Probably should add `use dwrote::TextAnalysisSourceMethods`.